Things To Do/Utilities/UserDefaultsHelper.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  UserDefaultsHelper.swift
+//  Things To Do
+//
+//  Created by Feyyaz ONUR on 2.07.2024.
+//
+
+import Foundation
+
+protocol UserDefaultsHelperProtocol {
+    associatedtype T: Codable
+    func saveData(_ items: [T])
```

Comment:
.

Reply:
when i change [T] to T, it gives me error at ViewModel like "Cannot convert return expression of type 'ToDoItem' to return type '[ToDoItem]'". so i guess T does not include [T]. and i found a usage of [T] in swift docs: https://docs.swift.org/swift-book/documentation/the-swift-programming-language/generics/